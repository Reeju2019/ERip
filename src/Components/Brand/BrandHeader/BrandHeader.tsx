import { Col, Container, Image, Row } from 'react-bootstrap'
import './BrandHeader.css'
import ProductData from '../../../Data/Product.mock.json'
import React, { useEffect, useState } from 'react'

interface Device {
  device_name: string
  device_top_img: string
  device_top_text1: string
  device_top_text2: string
  brand: {
    brand_name: string
    brand_image: string
    model: {
      model_name: string
      image: string
    }[]
  }[]
}

const BrandHeader: React.FunctionComponent = () => {
  const [selectedDevice, setSelectedDevice] = useState<Device>()
  useEffect(() => {
    const device: string = localStorage.getItem('deviceId') as string
    const deviceId: string = JSON.parse(device)
    ProductData.device.map((deviceItem) => {
      deviceItem.device_name === deviceId ? setSelectedDevice(deviceItem) : null;
    })
  }, [])

  return (
    <>
      <Container fluid>
        <Row
          className='brand-header'
          style={{ backgroundImage: 'url(/static/brandPageImage/bg-brand.png)' }}
        >
          <Col className='d-flex justify-content-center align-self-center align-item-center'>
            <h1 className='text-white'>
              {selectedDevice?.device_top_text1} <br />
              {selectedDevice?.device_top_text2}
            </h1>
          </Col>
          <Col className='d-flex justify-content-end'>
            <Image src={selectedDevice?.device_top_img} width='48%' />
          </Col>
        </Row>
      </Container>
    </>
  )
}

export default BrandHeader
