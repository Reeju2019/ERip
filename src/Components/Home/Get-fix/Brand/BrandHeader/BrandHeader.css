.brand-header {
  background-image: url(../../../../../../public/static/brandPageImage/bg-brand.png);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  
}