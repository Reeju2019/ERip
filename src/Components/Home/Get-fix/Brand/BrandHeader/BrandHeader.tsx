import { Col, Container, Image, Row } from 'react-bootstrap'
import './BrandHeader.css'

const BrandHeader: React.FunctionComponent = () => {
  return (
    <>
      <Container fluid>
        <Row className='brand-header mt-2'>
          <Col className='d-flex justify-content-center align-self-center align-item-center'>
            <h1 className='text-white'>
              On-site Mobile Repare <br /> Service in 30 Minutes
            </h1>
          </Col>
          <Col className='d-flex justify-content-end'>
            <Image src='/static/brandPageImage/Mobile.png' width='48%' />
          </Col>
        </Row>
      </Container>
    </>
  )
}

export default BrandHeader
