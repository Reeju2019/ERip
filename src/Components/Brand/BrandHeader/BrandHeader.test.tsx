import { render, screen } from '@testing-library/react'
import BrandHeader from './BrandHeader'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'

test('brand header', () => {
  render(<BrandHeader />)
})

describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<BrandHeader />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<BrandHeader />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<BrandHeader />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
