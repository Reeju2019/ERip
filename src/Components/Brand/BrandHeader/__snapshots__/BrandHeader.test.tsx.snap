// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
<div
  className="container-fluid"
>
  <div
    className="brand-header row"
    style={
      Object {
        "backgroundImage": "url(/static/brandPageImage/bg-brand.png)",
      }
    }
  >
    <div
      className="d-flex justify-content-center align-self-center align-item-center col"
    >
      <h1
        className="text-white"
      >
         
        <br />
      </h1>
    </div>
    <div
      className="d-flex justify-content-end col"
    >
      <img
        className=""
        width="48%"
      />
    </div>
  </div>
</div>
`;

exports[`App rendering specification SnapShot testing 2`] = `
<div
  className="container-fluid"
>
  <div
    className="brand-header row"
    style={
      Object {
        "backgroundImage": "url(/static/brandPageImage/bg-brand.png)",
      }
    }
  >
    <div
      className="d-flex justify-content-center align-self-center align-item-center col"
    >
      <h1
        className="text-white"
      >
         
        <br />
      </h1>
    </div>
    <div
      className="d-flex justify-content-end col"
    >
      <img
        className=""
        width="48%"
      />
    </div>
  </div>
</div>
`;
